File: backend/app/schemas/qr_code.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel

class QRCodeBase(BaseModel):
    name: str
    description: Optional[str] = None
    is_active: bool = True
    expiry_date: Optional[datetime] = None

class QRCodeCreate(QRCodeBase):
    pass

class QRCodeUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    is_active: Optional[bool] = None
    expiry_date: Optional[datetime] = None

class QRCode(QRCodeBase):
    id: str
    owner_id: str
    created_at: datetime
    description: Optional[str] = None
    is_active: bool = True
    expiry_date: Optional[datetime] = None

    class Config:
        from_attributes = True  # Allows conversion from SQLAlchemy model

File: frontend/lib/models/qr_code.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'qr_code.freezed.dart';
part 'qr_code.g.dart';

@freezed
class QRCode with _$QRCode {
  const factory QRCode({
    String? id,
    String? ownerId,
    required String name,
    String? description,
    @Default(true) bool isActive,
    DateTime? expiryDate,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) = _QRCode;

  factory QRCode.fromJson(Map<String, dynamic> json) => _$QRCodeFromJson(json);
}
